### PR TITLE
Fix register handler with different HTTP method

### PR DESCRIPTION
**Pull request by cheungpat**
This fixes the bug when using multiple handler decorators to register
the same handler path but with different HTTP method.

connects #164

### CODE REVIEW

File: skygear/registry.py [hunk 1]
```diff
@@ -57,16 +58,47 @@ def __init__(self):
...
         self.exception_handlers = {}
 
+    def _add_param_handler(self, param):
+        new_param_list = []
+
+        # Check existing param. Remove existing param if it overlaps with
+        # the one being registered.
+        for old_param in self.param_map['handler']:
+            if not old_param.get('name') == param.get('name'):
+                continue
```
Comment (carmenlau) [inline skygear/registry.py]: Should add `old_param` to `new_param_list` in this case

File: skygear/tests/test_registry.py [hunk 2]
```diff
@@ -35,11 +35,49 @@ def handler():
...
+        assert registry.get_handler('plugin:handler', 'POST') == handler2
+        assert registry.get_handler('plugin:handler', 'PUT') is None
+        assert registry.get_handler('plugin:handler3', 'POST') == handler3
+
+        param_map = registry.param_map['handler']
+        print(param_map)
+        assert len(param_map) == 3
+        assert param_map[0]['name'] == 'plugin:handler'
+        assert param_map[1]['name'] == 'plugin:handler'
+        assert 'POST' in param_map[1]['methods']
```
Comment (rickmak) [inline skygear/tests/test_registry.py]: I think what we want to test it `assert 'POST' not in param_map[0]['methods']`.

File: skygear/tests/test_registry.py [hunk 3]
```diff
@@ -35,11 +35,49 @@ def handler():
...
+        registry.register_handler('plugin:handler3', handler3, method='POST')
+
+        assert len(registry.handler) == 2
+        assert registry.get_handler('plugin:handler', 'GET') == handler1
+        assert registry.get_handler('plugin:handler', 'POST') == handler2
+        assert registry.get_handler('plugin:handler', 'PUT') is None
+        assert registry.get_handler('plugin:handler3', 'POST') == handler3
+
+        param_map = registry.param_map['handler']
+        print(param_map)
```
Comment (rickmak) [inline skygear/tests/test_registry.py]: No print?

Comment (cheungpat) [reply skygear/tests/test_registry.py]: of course

**Pull request merged**